Source/Applications/openXDA/openXDA/Controllers/Widgets/EventWidgetViewController.cs

```diff
@@ -110,60 +112,105 @@ public IHttpActionResult EventCountByMonth([FromBody] JObject query)
 
                 string meterFilter = postData.MeterIDs.Any() ? $" AND Meter.ID IN ({string.Join(",", postData.MeterIDs)})" : "";
 
+                // Unit we aggregate in
+                string aggUnit = null;
+                // Extra unit for aggregation if we need finer grain
+                string extraAggUnit = null;
+                // Unit to return aggregation in, useful to sometimes be different from aggregation unit due to weeks being a headache to deal with
+                string aggReturnUnit = null;
+                // Timespan for breaking up total time
+                string timeSpan;
+                switch (postData.Granularity.ToUpper())
+                {
+                    case "HOURLY":
+                        aggUnit = "DAYOFYEAR";
+                        extraAggUnit = "HOUR";
+                        timeSpan = "HOUR";
+                        break;
+                    default:
+                    case "MONTHLY":
+                        aggUnit = "MONTH";
+                        timeSpan = "MONTH";
+                        break;
+                    // Special notes about weekly, the first week and last week of the year may be less than 7 days due to cutoffs
+                    case "WEEKLY":
+                        aggUnit = "WEEK";
+                        aggReturnUnit = "DAYOFYEAR";
+                        timeSpan = "WEEK";
+                        break;
+                    case "DAILY":
+                        aggUnit = "DAYOFYEAR";
+                        timeSpan = "DAY";
+                        break;
+                    case "YEARLY":
+                        timeSpan = "YEAR";
+                        break;
+                }
```

Comment:
This seems overly complicated.
Could we just add a Column using `FORMAT( value , format [ , culture ] )` and use an apporpriate `format` like `YYYY` for year, `YYYY-MM` for month, `YYYY-MM-DD` for day, `YYYY-MM-DD-HH` for hour etc